src/System.CommandLine/Parsing/CommandResultExtensions.cs

```diff
@@ -36,7 +36,7 @@ public static class CommandResultExtensions
             {
                 foreach (var argument in commandResult.Command.Arguments)
                 {
-                    if (valueName.IsMatch(argument.Name))
+                    if (valueName.IsMatch(argument.Name) || valueName.IsMatch(argument.Name.Replace("|","or")))
```

Comment:
This change should maybe be moved to the `IsMatch` method.

Reply:
Done. I had considered not doing to before to allow the or with and without the replacement. After realising that a `|` is invalid in a parameter name - this shouldn't cause any breaking changes.